src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp

```diff
@@ -815,6 +815,7 @@ class boss_the_lich_king : public CreatureScript
                 {
                     me->GetMap()->SetZoneOverrideLight(AREA_ICECROWN_CITADEL, LIGHT_SNOWSTORM, 5000);
                     me->GetMap()->SetZoneWeather(AREA_ICECROWN_CITADEL, WEATHER_STATE_LIGHT_SNOW, 0.5f);
+                    summons.DespawnEntry(NPC_SHADOW_TRAP);
```

Comment:
Naah, you are correct.
Both spells is Self cast with SPELL_AURA_PERIODIC_TRIGGER_SPELL  effect
It will hit only Lich King, so will be called 1 time per cast